vignettes/deepdive.Rmd

```diff
@@ -0,0 +1,404 @@
+
+```{r}
+adsl_lbl <- xportr_label(.df = adsl_lbl, metadata = var_spec_lbl, domain = "ADSL", verbose = "warn")
+```
+
+Success! All labels have been applied that are present in the both the metadata and the dataset. However, please note that the `TRTSDT` variable has had the label with characters greater than 40 **applied** to the dataset and the `TRTDUR` and `DCREASCD` have empty variable labels.  
+
+```{r, max.height='300px', attr.output='.numberLines', echo = FALSE}
+str(adsl_lbl)
+```
+
+Just like we did for the other functions, setting `verbose = "stop"` immediately stops R from processing the labels. Here the function detects the mismatches between the variables and labels as well as the label that is greater than 40 characters. As this stops the process, none of the labels will be applied to the dataset until corrective action is applied.
+
+```{r, echo = TRUE, error = TRUE}
+adsl_label <- xportr_label(.df = adsl_lbl, metadata = var_spec_lbl, domain = "ADSL", verbose = "stop")
+```
+
+## `xportr_order()`
+
+The order of the dataset can greatly increase readability of the dataset for downstream stakeholders.  For example, having all the treatment related variables or analysis variables grouped together can help with inspection and understanding of the dataset. `xportr_order()` can take the order information from the metadata and apply it to your dataset.
+
+```{r}
+adsl_ord <- xportr_order(adsl, var_spec, "ADSL", verbose = "warn")
+```
+
+Readers are encouraged to inspect the dataset and metadata to see the past order and updated order after calling the function. Note the messaging from `xportr_order()`:
+
+* Variables not in the metadata are moved to the end
+* Variables not in order are re-ordered and a message is printed out on which ones were re-rordered.
+
+
+```{r, echo = TRUE, error = TRUE}
+adsl_ord <- xportr_order(.df = adsl, metadata = var_spec, domain = "ADSL", verbose = "stop")
+```
+
+Just like we did for the other functions, setting `verbose = "stop"` immediately stops R from processing the order. If variables or metadata are missing from either, the re-ordering will not process until corrective action is performed.
+
+## `xportr_format()`
+
+Formats play an important role in the SAS language and have a column in specification files. Being able to easily apply formats into your `xpt` file will allow downstream users of SAS to quickly format the data appropriately when reading into a SAS-based system.  `xportr_format()` can take these formats and apply them. Please reference `xportr_length()` or `xportr_label()` to note the missing `attr()` for formats in our `ADSL` dataset.
+
+This example is slightly different from previous examples. You will need to use `xportr_type()` to coerce R Date variables and others types to character or numeric. Only then can you use `xportr_format()` to apply the format column to the dataset.
+
```

Comment:
I actually think if the variable is a Date then `xportr_format()` won't apply the format

Reply:
Here is example of `xportr_format()` not applying formats to the dataframe as a single call if the variables are of the Date class.  If you use `xportr_format()` with `xportr_type()` then it applies the formats

@atorus-research/xportr-development-team not sure if this was intentionally designed like this, but we should make a note in the documentation that these two are needed.  If the variables is already a character then you don't need to use ``xportr_type()1

```
library(lubridate)

adsl <- tribble(
  ~STUDYID,   ~USUBJID,              ~TRTEDTM,      ~TRTEDT,
  "PILOT01", "01-1130", "2014-08-16 23:59:59", "2014-08-16",
  "PILOT01", "01-1133", "2013-04-28 23:59:59", "2013-04-28",
  "PILOT01", "01-1211", "2013-01-12 23:59:59", "2013-01-12",
  "PILOT01", "09-1081", "2014-04-27 23:59:59", "2014-04-27",
  "PILOT01", "09-1088", "2014-10-09 23:59:59", "2014-10-09"
) %>% 
  mutate(TRTEDTM = ymd_hms(TRTEDTM, tz=Sys.timezone()),
         TRTEDT = as.Date(TRTEDT),
         STUDYID = as.factor(STUDYID))

metadata <- data.frame(
  dataset = c("adsl", "adsl", "adsl", "adsl"),
  variable = c("STUDYID", "USUBJID", "TRTEDTM", "TRTEDT"),
  type = c("character", "character", "numeric", "numeric"),
  format = c(NA, NA, "DATETIME20.", "DATE9.")
)

adsl_notype <- xportr_format(adsl, metadata)

adsl_type <- adsl %>% 
  xportr_type(metadata = metadata) %>% 
  xportr_format(metadata = metadata)

```

![image](https://github.com/atorus-research/xportr/assets/10111024/001df3c4-d56b-4e85-8f98-4775c9a620fc)

Reply:
Thanks @bms63 for the example code.

I also ran `xportr_write(adsl_type, "typefmt.xpt") ` and open it in SAS. As the date class is coerced to dbl, the date is incorrect  in SAS:
![image](https://github.com/atorus-research/xportr/assets/69685640/4ec065ad-382a-4f2e-9090-452a76f9e17a)


Date class will not be coerced anymore after the update  #145 to avoid incorrect date in SAS.
I tested with the example code in the branch of the PR #145 and the format is not added in adsl_type. 
I guess, we will need to update `xportr_format()`, right?